csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsources/Stored.qll

```diff
@@ -27,30 +37,30 @@ class DbRawSqlStoredFlowSource extends StoredFlowSource {
  * An expression that has a type of `DbDataReader` or a sub-class, representing the result of a
  * data command.
  */
-class DbDataReaderStoredFlowSource extends StoredFlowSource {
+class DbDataReaderStoredFlowSource extends DatabaseInputSource {
   DbDataReaderStoredFlowSource() {
     this.asExpr().getType() = any(SystemDataCommon::DbDataReader dataReader).getASubType*()
   }
 }
 
 /** An expression that accesses a method of `DbDataReader` or a sub-class. */
-class DbDataReaderMethodStoredFlowSource extends StoredFlowSource {
+class DbDataReaderMethodStoredFlowSource extends DatabaseInputSource {
   DbDataReaderMethodStoredFlowSource() {
     this.asExpr().(MethodCall).getTarget().getDeclaringType() =
       any(SystemDataCommon::DbDataReader dataReader).getASubType*()
   }
 }
 
 /** An expression that accesses a property of `DbDataReader` or a sub-class. */
-class DbDataReaderPropertyStoredFlowSource extends StoredFlowSource {
+class DbDataReaderPropertyStoredFlowSource extends DatabaseInputSource {
   DbDataReaderPropertyStoredFlowSource() {
     this.asExpr().(PropertyAccess).getTarget().getDeclaringType() =
       any(SystemDataCommon::DbDataReader dataReader).getASubType*()
   }
 }
 
 /** A read of a mapped property. */
-class ORMMappedProperty extends StoredFlowSource {
+class ORMMappedProperty extends DatabaseInputSource {
   ORMMappedProperty() {
     this instanceof EntityFramework::StoredFlowSource or
```

Comment:
Maybe we should consider to let EntityFramework::StoredFlowSource extends Stored::StoredFlowSource instead (the same for NHibernate::StoredFlowSource). Then we have the same pattern like for java for different frameworks.
Then OrmMappedProperty should just extend `DataFlow::Node` (is it used anywhere? If not we can consider to deprecate it).

Reply:
Converting `EntityFramework::StoredFlowSource` and `NHibernate::StoredFlowSource` to extend `DatabaseInputSource` makes sense. They are also defined in ways where as I understand it, they cannot be converted to models-as-data.

`ORMMappedProperty` is not used anywhere directly within our codebase or from what I can find on https://github.com/search. I think it would be reasonable to deprecate it.